### PR TITLE
[JUJU-1652, JUJU-1948] API client resources to gomock

### PR DESCRIPTION
**Pull request by basebandit**
Slight clean up of some api/client/resources tests.

## Checklist

*If an item is not applicable, use `~strikethrough~`.*

- [x] Code style: imports ordered, good names, simple structure, etc
- ~[ ] Comments saying why design decisions were made~
- [x] Go unit tests, with comments saying what you're testing
- ~[ ] [Integration tests](https://github.com/juju/juju/tree/develop/tests), with comments saying what you're testing~
- ~[ ] [doc.go](https://discourse.charmhub.io/t/readme-in-packages/451) added or updated in changed packages~

## QA steps

All unit tests should pass.

### CODE REVIEW

**Review by cderici — APPROVED** [review]
Nice cleanup. LGTM.

File: api/client/cloud/cloud_test.go [hunk 1]
```diff
@@ -811,28 +811,15 @@ func (s *cloudSuite) TestUpdateCloudsCredentialsManyMatchingResults(c *gc.C) {
...
-
 	res := new(params.UpdateCredentialResults)
 	results := params.UpdateCredentialResults{
 		Results: []params.UpdateCredentialResult{
 			{},
 			{},
 		}}
 	mockFacadeCaller := basemocks.NewMockFacadeCaller(ctrl)
-	mockFacadeCaller.EXPECT().FacadeCall("UpdateCredentialsCheckModels", args, res).SetArg(2, results).Return(nil)
+	mockFacadeCaller.EXPECT().FacadeCall("UpdateCredentialsCheckModels", gomock.Any(), res).SetArg(2, results).Return(nil)
```
Comment (hmlanigan) [inline api/client/cloud/cloud_test.go:822]: this removes most of the content of this test. suggest you using a go mock matcher instead

**Review by hmlanigan — CHANGES_REQUESTED** [review]

File: api/client/cloud/cloud_test.go (hunk 1 above)
Comment (basebandit) [reply api/client/cloud/cloud_test.go]: Oh yeah thanks this might be better. Putting one right away.

File: api/client/cloud/cloud_test.go [hunk 2]
```diff
@@ -908,3 +922,25 @@ func (s *cloudSuite) TestAddCloudsCredentials(c *gc.C) {
...
+	cred, ok := x.(params.UpdateCredentialArgs)
+	if !ok {
+		return false
+	}
+	if len(cred.Credentials) != len(c.arg.Credentials) {
+		return false
+	}
+	if cred.Force != c.arg.Force {
+		return false
+	}
```
Comment (hmlanigan) [inline api/client/cloud/cloud_test.go:940]: you can also add checking of the c.arg.Credential contents.

Comment (basebandit) [reply api/client/cloud/cloud_test.go:954]: Sure thing.

**Review by hmlanigan — APPROVED** [review]

**Pull request merged**